### ISSUE TITLE
Cannot get the newly created tag in postpublish hook.

### ISSUE THREAD

**Issue by lxzxl** [yarnpkg/yarn#5243]
- Yarn - v1.4.0
- Node - v8.9.3
- macOS 10.13.2

Images below can explain my issue: In `postpublish` hook, it cannot get the newly created tag by publish command in  I'm not sure if this is intended to do so. I want to push the 

![image](https://user-images.githubusercontent.com/2275002/35082926-524580ec-fc42-11e7-9c97-445fa7ea2302.png)

![image](https://user-images.githubusercontent.com/2275002/35082820-d2da99f0-fc41-11e7-952c-00cafb7b6346.png)

**Comment by lelandmiller** [comment]
I am experiencing this with Yarn 1.3.2, Node 8.9.4, and macOS 10.12.6 if it helps eliminate anything.

**Comment by lelandmiller** [comment]
It seems this may be due to the commit happening after the full publish lifecycle is run (<https://github.com/yarnpkg/yarn/blob/aa200e4d63f8720738ca4c07bd5a38c2f89a402f/src/cli/commands/publish.js#L148-L149>). Is this the expected behavior for yarn, or a bug?

**Comment by zavr-1** [comment]
yeah it's annoying there's no way to automatically push the tag on publish, and I've forgotten to do that a couple of times on a PC to which I have to access now, therefore the version commits are probably lost forever.

**Comment by arlo-egg** [comment]
I think that `postversion` is run after the tag has been created: https://github.com/yarnpkg/yarn/blob/aa200e4d63f8720738ca4c07bd5a38c2f89a402f/src/cli/commands/version.js#L161

I have used the `postversion` hook to update tags after publishing as follows. However I think it would be more intuitive if the lifecycle hook to use in this situation was `postpublish`.
```
  "scripts": {
    "prepublish": "yarn test",
    "postversion": "git push --follow-tags",
  },
```

**Comment by cmdcolin** [comment]
@arlo-egg random note but your comment helped me a bunch. we had git push --follow-tags in postpublish and it never worked but I found after following your comment that it does work in postversion...weird but thank you!
